### DISCUSSION TITLE
Too many outputs from neighbouring_spins_sum

### DISCUSSION THREAD

**Discussion by benpymer** [Part-II-Computational-Physics/.github#56] [General]
Hi! I was defining neighbouring_spins_sum like this

'''
def neighbouring_spins_sum(i,j,lattice):
    '''Sums the spins of all neighbours of the spin at [i,j].'''
    
    # width is an unnecessary input to the function as we already have lattice in the inputs, so I have defined the function without it and 
    # calculated it inside. I also define an empty variable sum to contain the sum of the spins
    width = np.size(lattice, 0)
    sum = 0

    # finds the value of the lattice at each of the 4 adjacent sites returned by neighbouring_sites, and adds each one to sum
    for k in range(4):
        sum += lattice[neighbouring_sites(i,j,width)[k]]

    return sum

print(neighbouring_spins_sum(1,3,init_lattice(5,1)))
print(neighbouring_spins_sum(3,2,init_lattice(5,0)))
print(neighbouring_spins_sum(4,3,init_lattice(5,-1)))

'''

I was then trying to test it by printing some values as seen below for different types of lattice. My output from neighbouring_sites is a list of the 4 co-ordinates adjacent to the site (i.e should look like [[i-1,j] , [i+1, j], [i, j-1], [i, j+1]]) which is working as expected. I was hoping that I'd be able to index the lattice values that I wanted for my sum inside the for loop by passing each of the 4 elements of this list in turn, however on doing this, I seem to end up with a 2x5 array of 10 values, each of which makes sense for its lattice type (i.e the array for lattice type 1 is an array of ten 4s, for type 0 is an array of 10 random numbers between -4 and 4 and for type -1 is an array of ten -4s). On experimenting, I discovered that the size and shape of the output arrays are independent of the matrix width, i.e I get a 2x5 array for all widths. I tried using np.sum() instead of the for loop, but it told me that I had tried to pass 4 indices to a 2D array (I think it was referring to lattice as the 2D array, but I'm not sure. The error was definitely thrown on that line). What am I doing wrong here?

**Comment by Jamiewhbee** [comment]
What your code does is summing lattice[[i,j]] rather than lattice[i,j].
You probably want to unpack each element of the array produced by the neighbouring_sites function, e.g. define x,y = neighbouring,sites(i,j,width[k]), then take the sum of lattice[x,y}.

**Reply by benpymer** [reply]
Thanks! All working now!

**Reply by benpymer** [reply]
Out of interest, what is lattice[[i,j]]? Not really sure what the double brackets do

**Reply by Jamiewhbee** [reply]
I think the output of lattice[[i,j]] is a matrix with the first row as the ith row of the lattice and the second row as the jth row.
